osquery/tables/system/windows/system_util.cpp

```diff
@@ -0,0 +1,176 @@
+                          nullptr) == 0) {
+    return std::string("");
+  }
+
+  return std::string(buffer.begin(), buffer.end());
+}
+
+WmiResultItem::WmiResultItem(WmiResultItem&& src) {
+  result_ = nullptr;
+  std::swap(result_, src.result_);
+}
+
+WmiResultItem::~WmiResultItem() {
+  if (result_ != nullptr) {
+    result_->Release();
+    result_ = nullptr;
+  }
+}
+
+void WmiResultItem::PrintType(const std::string& name) const {
+  std::wstring property_name = string_to_wstring(name);
+
+  VARIANT value;
+  HRESULT hr = result_->Get(property_name.c_str(), 0, &value, nullptr, nullptr);
+  if (hr != S_OK) {
+    std::cerr << "Failed: " << name << "\n";
+  } else {
+    std::cout << "Name=" << name << ", Type=" << value.vt << "\n";
+    if (value.vt == VT_I4) {
+      std::cout << "  Value=" << value.lVal << "\n";
+    } else if (value.vt == VT_BSTR) {
+      std::wcout << "  Value=" << value.bstrVal << "\n";
+    }
+  }
+}
+
+long WmiResultItem::GetLong(const std::string& name) const {
+  std::wstring property_name = string_to_wstring(name);
+
+  VARIANT value;
+  HRESULT hr = result_->Get(property_name.c_str(), 0, &value, nullptr, nullptr);
+  if (hr != S_OK || value.vt != VT_I4) {
+    return -1;
+  }
+  return value.lVal;
+}
+
+std::string WmiResultItem::GetString(const std::string& name) const {
+  std::wstring property_name = string_to_wstring(name);
+
+  VARIANT value;
+  HRESULT hr = result_->Get(property_name.c_str(), 0, &value, nullptr, nullptr);
+  if (hr != S_OK || value.vt != VT_BSTR) {
+    return std::string("");
+  }
+  return BSTR_to_string(value.bstrVal);
+}
+
+WmiRequest::WmiRequest(const std::string& query) {
+  std::wstring wql = string_to_wstring(query);
+
+  HRESULT hr = E_FAIL;
+
+  hr = ::CoInitializeEx(0, COINIT_MULTITHREADED);
+  hr = ::CoInitializeSecurity(
+    nullptr, -1, nullptr, nullptr, RPC_C_AUTHN_LEVEL_DEFAULT,
+    RPC_C_IMP_LEVEL_IMPERSONATE, nullptr, EOAC_NONE, nullptr);
+  hr = ::CoCreateInstance(CLSID_WbemLocator, 0, CLSCTX_INPROC_SERVER, IID_IWbemLocator, (LPVOID *)&locator_);
+  if (hr != S_OK) {
+    locator_ = nullptr;
+    return;
+  }
+
+  hr = locator_->ConnectServer(L"ROOT\\CIMV2", nullptr, nullptr, nullptr, 0, nullptr, nullptr, &services_);
+  if (hr != S_OK) {
+    services_ = nullptr;
+    return;
+  }
+
+  hr = services_->ExecQuery(L"WQL", (BSTR)wql.c_str(), WBEM_FLAG_FORWARD_ONLY, nullptr, &enum_);
+  if (hr != S_OK) {
+    enum_ = nullptr;
+    return;
+  }
+
+  IWbemClassObject *result = nullptr;
+  ULONG result_count = 0;
+
+  while (enum_->Next(WBEM_INFINITE, 1, &result, &result_count) == S_OK) {
+    results_.push_back(WmiResultItem(result));
+  }
+
+  status_ = true;
+}
+
+WmiRequest::WmiRequest(WmiRequest&& src) {
+  locator_ = nullptr;
+  std::swap(locator_, src.locator_);
+
+  services_ = nullptr;
+  std::swap(services_, src.services_);
+
+  enum_ = nullptr;
+  std::swap(enum_, src.enum_);
+}
+
+WmiRequest::~WmiRequest() {
+  results_.clear();
```

Comment:
Yep you do, WmiResultItem takes ownership of the pointer and calls `Release` when it goes out of scope. Going to add a comment to make it more clear, since I had to track it down as well.